treeherder/webapp/api/performance_data.py

```diff
@@ -492,3 +495,35 @@ def list(self, request):
 
         serializer = self.get_serializer(self.queryset, many=True)
         return Response(data=serializer.data)
+
+
+class ValidityDashboardViewSet(viewsets.ViewSet):
+
+    def list(self, request):
+        query_params = ValidityDashboardParamsSerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors,
+                            status=HTTP_400_BAD_REQUEST)
+
+        framework_id = query_params.validated_data['framework']
+
+        query_set_one = (PerformanceSignature.objects
+                         .prefetch_related('performancealert')
+                         .values('suite', 'test')
+                         .annotate(repositories=GroupConcat('repository_id', distinct=True))
+                         .annotate(platforms=GroupConcat('platform_id', distinct=True))
+                         .annotate(total_alerts=Count('performancealert'))
+                         .filter(framework_id=framework_id)
```

Comment:
If data is not affected, please .filter(...) first and .annotate(...) after. It should improve performance. [Another guys](https://stackoverflow.com/questions/8743505/django-aggregate-or-annotate) struggled with this. Please, test this in order to see if the output is not changed.

Reply:
I tested with .filter() after .prefetch_related() and the data returned is the same. (URL used for test: localhost:5000/api/performance/validity-dashboard/?framework=10)
I think that the .filter() must be done right after after .prefetch_related(), because filtering before prefetching does not guarantee that the QuerySet has only items related to given framework_id.